sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -584,13 +579,8 @@ case class AlterTableSetLocationCommand(
 
 
 object DDLUtils {
-
-  def isDatasourceTable(props: Map[String, String]): Boolean = {
-    props.contains(DATASOURCE_PROVIDER)
-  }
-
   def isDatasourceTable(table: CatalogTable): Boolean = {
-    isDatasourceTable(table.properties)
+    table.provider.isDefined && table.provider.get != "hive"
```

Comment:
We are checking the same thing but using different ways. See the code: https://github.com/cloud-fan/spark/blob/96d57b665ac65750eb5c6f9757e5827ea9c14ca4/sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala#L175

Could we make them consistent? or leave a comment to explain it?